### ISSUE TITLE
feat: layer_forbidden_imports rule — ban crate imports per architectural layer

### ISSUE THREAD

**Issue by kgatilin** [mshogin/archlint#115]
## Use Case

Domain layer must not import `serde_json`, `chrono`, `tokio`. App layer should not import infra crates except at composition root. Current rules don't catch these domain impurities (e.g. `serde_json::Value` in types, `chrono::Utc::now()` calls).

## Proposed Config

```yaml
rules:
  layer_forbidden_imports:
    enabled: true
    level: taboo
    layers:
      domain:
        forbidden: ["serde", "serde_json", "tokio", "chrono"]
      ports:
        forbidden: ["tokio", "serde"]
```

## Expected Behavior

- For each layer, scan `use` statements and flag any import from a forbidden crate
- Report as violation at the configured level (taboo/telemetry)
